src/main/java/de/upb/crypto/math/structures/cartesian/GroupElementExpressionVector.java

```diff
@@ -0,0 +1,139 @@
+package de.upb.crypto.math.structures.cartesian;
+
+import de.upb.crypto.math.expressions.exponent.ExponentExpr;
+import de.upb.crypto.math.expressions.group.GroupElementExpression;
+import de.upb.crypto.math.expressions.group.GroupEmptyExpr;
+import de.upb.crypto.math.interfaces.structures.GroupElement;
+import de.upb.crypto.math.interfaces.structures.RingElement;
+import de.upb.crypto.math.pairings.generic.BilinearMap;
+import de.upb.crypto.math.serialization.ListRepresentation;
+import de.upb.crypto.math.serialization.Representable;
+import de.upb.crypto.math.serialization.Representation;
+import de.upb.crypto.math.structures.zn.Zn;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * A vector of expressions supporting various operations on its elements.
+ */
+public class GroupElementExpressionVector extends Vector<GroupElementExpression> {
+
+    public GroupElementExpressionVector(GroupElementExpression... values) {
+        super(values);
+    }
+
+    public GroupElementExpressionVector(List<GroupElementExpression> values) {
+        super(values);
+    }
+
+    public GroupElementExpressionVector(Vector<? extends GroupElementExpression> vector) {
+        this(vector.values, true);
```

Comment:
Accessing the member variable `values` like this won't work anymore after the math reorganization since `Vector` is now in a different package than the specific vector implementations.